packages/node-core/src/indexer/StoreOperations.test.ts

```diff
@@ -0,0 +1,201 @@
+      },
+      {
+        name: 'createdBlock',
+        type: 'Int',
+        isArray: false,
+        nullable: true,
+        isEnum: false,
+      },
+      {
+        name: 'lastEvent',
+        type: 'String',
+        isArray: false,
+        nullable: true,
+        isEnum: false,
+      },
+    ],
+    indexes: <any[]>[],
+  },
+];
+
+const option: DbOption = {
+  host: process.env.DB_HOST ?? '127.0.0.1',
+  port: process.env.DB_PORT ? Number(process.env.DB_PORT) : 5432,
+  username: process.env.DB_USER ?? 'postgres',
+  password: process.env.DB_PASS ?? 'postgres',
+  database: process.env.DB_DATABASE ?? 'postgres',
+  timezone: 'utc',
+};
+
+jest.setTimeout(10_000);
+
+describe('StoreOperations with db', () => {
+  let sequelize: Sequelize;
+  let schema: string;
+  let model: any;
+  let cacheModel: CachedModel<EraEntity>;
+
+  const flush = async (blockHeight: number) => {
+    const tx = await sequelize.transaction();
+    await cacheModel.flush(tx, blockHeight);
+    await tx.commit();
+  };
+
+  beforeAll(async () => {
+    process.env.TZ = 'UTC';
+    sequelize = new Sequelize(
+      `postgresql://${option.username}:${option.password}@${option.host}:${option.port}/${option.database}`,
+      option
+    );
+    await sequelize.authenticate();
+
+    schema = '"storeOp-test-1"';
+    await sequelize.createSchema(schema, {});
+
+    const modelFactory = sequelize.define(
+      'eraEntity',
+      {
+        id: {
+          type: String.sequelizeType,
+          primaryKey: true,
+        },
+        startTime: {type: DateObj.sequelizeType},
+        endTime: {type: DateObj.sequelizeType, allowNull: true},
+        lastEvent: {type: String.sequelizeType, allowNull: true},
+        forceNext: {type: Boolean.sequelizeType, allowNull: true},
+        createdBlock: {type: Int.sequelizeType, allowNull: true},
+      },
+      {timestamps: false, schema: schema}
+    );
+    model = await modelFactory.sync().catch((e) => {
+      console.log('error', e);
+      throw e;
+    });
+
+    let i = 0;
+
+    cacheModel = new CachedModel(model, false, new NodeConfig({} as any), () => i++);
+  });
+
+  afterAll(async () => {
+    await sequelize.dropSchema(schema, {logging: false});
+    await sequelize.close();
+  });
+
+  it('check operation stack consistency with data fetched from db and cache', async () => {
+    const operationStackDb = new StoreOperations(models);
+    const operationStackCache = new StoreOperations(models);
+
+    cacheModel.set(
+      `0x01`,
+      {
+        id: `0x01`,
+        startTime: new Date('2024-02-14T02:38:51.000Z'),
+        forceNext: false,
+        createdBlock: 10544492,
+      },
+      1
+    );
+    await flush(2);
+
+    const rawDb = (await cacheModel.model.findOne({where: {id: '0x01'}}))?.toJSON();
+
+    const rawCache = await (cacheModel as any).getCache.get('0x01');
+
+    const entityDb = EntityClass.create<EraEntity>('EraEntity', rawDb, {} as Store);
+    if (entityDb) {
```

Comment:
Wont this always be true?

Reply:
https://github.com/subquery/subql/blob/1cfae76b3534f6541f0c04a09416b79abafdd2ef/packages/node-core/src/indexer/store/entity.ts#L18

it should be true, but need this for return type assertion

Reply:
I was getting confused with this not being undefined.https://github.com/subquery/subql/pull/2495/files#diff-46a6558964a1655301f62eff47b0ce02ce083d8500fbec26a8c446ce8f4eb423R28

I think there should be an expect here instead. If both entities are undefined then the operationStacks aren't updated and will always be the same, then this could pass when it shouldn't